lib/ip.js

```diff
@@ -376,7 +376,25 @@ ip.address = function(name, family) {
     return res[0].address;
   }
 
-  var all = Object.keys(interfaces).map(function (nic) {
+  function priority (name) {
+    if (name.startsWith('en') || name.startsWith('eth')) {
+      return 0;
+    }
+    if (name.startsWith('wlan')) {
+      return 1;
+    }
+
+    return 2;
+  }
+
+  var sortedInterfaces = Object.keys(interfaces).sort(function(a, b) {
+    a = priority(a);
+    b = priority(b);
+
+    return a - b;
```

Comment:
why not just `return priority(a) - priority(b)`?

Reply:
@brentvatne you're right :-) I did that to debug my code.